### PR TITLE
Issue #10144: fix classes count on multicatch and implements clause

### LINKED ISSUE #10144: ClassFanOutComplexity fails to count all classes in multicatch and implements clause
I have read check documentation: https://checkstyle.sourceforge.io/config_metrics.html#ClassFanOutComplexity
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```

➜  src javac Test.java
Note: Test.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

➜  src cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="ClassFanOutComplexity">
          <property name="max" value="1"/>
<!--          <property name="memberModifiers" value="LITERAL_PUBLIC, FINAL"/>-->
        </module>
    </module>
</module>

➜  src cat Test.java
import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.util.EmptyStackException;

public class Test {
    public static void main(String[] args) {
        try {
            System.out.println(args[7]);
            File myFile = new File("myfile.txt"); // 1

            InputStream stream = myFile.toURL().openStream(); // 2
        } catch (IOException  | EmptyStackException e) { // 3,4

        }
    }

}

➜  src java -jar checkstyle-8.43-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/Desktop/Tester/src/Test.java:6:1: Class Fan-Out Complexity is 3 (max allowed is 1). [ClassFanOutComplexity]
Audit done.
Checkstyle ends with 1 errors.
➜  src 


```

I would expect the "Class Fan-Out Complexity" to be 4.

### PR DESCRIPTION
**Pull request by strkkk**
Closes #10144

Diff Regression config: https://gist.githubusercontent.com/strkkk/31749e0743553a606f69fdfc6670d780/raw/f555bc03ede3366afdd92cfe337d12565591b96a/confg.xml

TODO items
- [x] fix xwiki issue https://app.circleci.com/pipelines/github/checkstyle/checkstyle/9240/workflows/0f48c81d-be8b-4871-8c4f-a5c7a957623f/jobs/102304 - https://jira.xwiki.org/browse/XCOMMONS-2294
- [x] fix xwiki platform issue https://app.circleci.com/pipelines/github/checkstyle/checkstyle/9251/workflows/6c58e0bd-c81f-4bb1-82f3-c87920616f1e/jobs/102361 - https://jira.xwiki.org/browse/XWIKI-19055

### CODE REVIEW

**Comment by strkkk** [comment]
Github, generate report

File: config/checkstyle_checks.xml [hunk 1]
```diff
@@ -660,7 +660,8 @@
                        FileNotFoundException, TestException, Log, Sets, Multimap,
                        TokenStreamRecognitionException, RecognitionException,
                        TokenStreamException, IOException, Override, Deprecated, SafeVarargs,
-                       SuppressWarnings, FunctionalInterface"/>
+                       SuppressWarnings, FunctionalInterface, NoSuchMethodException,
+                       InvocationTargetException, IllegalAccessException"/>
```
Comment (strkkk) [inline config/checkstyle_checks.xml:664]: this is necessary because there is a violation in AutomaticBean class, there are multicatch for these exceptions

**Comment by strkkk** [comment]
Github, rebase

**Comment by strkkk** [comment]
I have created new issue for xwiki https://jira.xwiki.org/browse/XWIKI-19055, will update head commit hash after they fix issues

**Comment by strkkk** [comment]
xwiki and other issues are resolved now

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/metrics/classfanoutcomplexity/InputClassFanOutComplexityImplementsAndNestedCount.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/*
+ClassFanOutComplexity
+max = 0
+excludedClasses = (default)ArrayIndexOutOfBoundsException, ArrayList, Boolean, Byte, \
```
[the remark this answers is not in the document]
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/metrics/classfanoutcomplexity/InputClassFanOutComplexityImplementsAndNestedCount.java:4]: @strkkk , I pushed changes to Inputs that tried to request from you.

Please take a look.
it is example how violation message shows details that never become outdated. And you do not need scroll to test to see a complexity value.
Comment that you did most likely eventually be outdated and confusing.

**Review by romani — APPROVED** [review]
ok to merge.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/metrics/classfanoutcomplexity/InputClassFanOutComplexityImplementsAndNestedCount.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (strkkk) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/metrics/classfanoutcomplexity/InputClassFanOutComplexityImplementsAndNestedCount.java:4]: ok, got it. I removed my comment and slightly modified test inputs

**Review by rnveach — APPROVED** [review]

**Review by rnveach — COMMENTED** [review]
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/b23de12_2021100138/reports/diff/checkstyle/index.html#A1
I am not seeing anything like `catch (IOException  | EmptyStackException e) {`. Why did counts change?

**Comment by romani** [comment]
It is case of "Same issue is for several interfaces in implements clause" that also fixed by this update

**Comment by romani** [comment]
As there are some uncertainty, let's do one more review

**Comment by rnveach** [comment]
Issue is only talking about multi-catches by report and title. There is a conflict then between the update and the issue which will also impact the release notes. This issue is marked as a bug.

**Comment by romani** [comment]
@strkkk , please update issue description and commit to be less confusing

**Comment by strkkk** [comment]
@romani @rnveach I update title and commit message

**Review by pbludov — APPROVED** [review]

**Comment by rnveach** [comment]
@pbludov I wasn't really finished reviewing the regression as I was waiting for confirmation on the change.
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/b23de12_2021100138/reports/diff/checkstyle/index.html#A3
Why did this one decrease?

**Comment by pbludov** [comment]
>  Why did this one decrease? That doesn't fit into the issue's description either.

I checked this file locally:
```
[WARN] tmp/TestClass.java:94:1: Class Fan-Out Complexity is 179 (max allowed is 20). [ClassFanOutComplexity]
```

It is not changed. Maybe there is an issue with report generation.

**Comment by romani** [comment]
@rnveach , 

> I wasn't really finished reviewing the regression as I was waiting for confirmation on the change.

Please do not approve PR until you 100% satisfied

**Comment by strkkk** [comment]
> https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/aea7fa1_2021115743/reports/diff/index.html

@rnveach you are looking at the old report, which is incorrect. The latest correct report is in qoute

**Comment by rnveach** [comment]
Ok, I was looking at an old report.

I thought I removed my approval since I forgot to look at the difference report.
> rnveach approved these changes 2 days ago
@rnveach rnveach self-requested a review 2 days ago 

I have no more issues now.

**Pull request merged**